src/bin/sol-fbp-runner/web-inspector.c

```diff
@@ -0,0 +1,1396 @@
+/*
+ * This file is part of the Soletta Project
+ *
+ * Copyright (C) 2015 Intel Corporation. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <inttypes.h>
+#include <stdio.h>
+#include <sys/stat.h>
+
+#include "sol-flow.h"
+#include "sol-flow-inspector.h"
+#include "sol-http-server.h"
+#include "sol-json.h"
+#include "sol-mainloop.h"
+#include "sol-util-file.h"
+#include "sol-util.h"
+#include "sol-util-internal.h"
+
+#include "web-inspector.h"
+
+static struct sol_http_server *server;
+static struct sol_http_progressive_response *events_response;
+
+static struct timespec start;
+
+static int
+web_inspector_add_json_key_value(struct sol_buffer *buf, const char *k, const char *v)
+{
+    int r;
+
+    r = sol_buffer_append_printf(buf, "\"%s\":", k);
+    if (r < 0)
+        return r;
+
+    return sol_json_serialize_string(buf, v);
+}
+
+static int
+web_inspector_add_key_value_array(struct sol_buffer *buf, const struct sol_vector *vector)
+{
+    uint16_t i;
+    const struct sol_key_value *kv;
+    int r;
+
+    SOL_VECTOR_FOREACH_IDX (vector, kv, i) {
+        if (i > 0) {
+            r = sol_buffer_append_char(buf, ',');
+            if (r < 0)
+                return r;
+        }
+
+        r = sol_buffer_append_char(buf, '[');
+        if (r < 0)
+            return r;
+
+        r = sol_json_serialize_string(buf, kv->key);
+        if (r < 0)
+            return r;
+
+        r = sol_buffer_append_char(buf, ',');
+        if (r < 0)
+            return r;
+
+        r = sol_json_serialize_string(buf, kv->value);
+        if (r < 0)
+            return r;
+
+        r = sol_buffer_append_char(buf, ']');
+        if (r < 0)
+            return r;
```

Comment:
Don't you think it's worth log this error ? Don't need to be here, can be in the top leve function, e. g. `web_inspector_will_deliver_packet`. As far as i could see the error is not spotted.